web/src/topic_link_util.ts

```diff
@@ -0,0 +1,63 @@
+import assert from "minimalistic-assert";
+
+import * as internal_url from "../shared/src/internal_url";
+
+import * as stream_data from "./stream_data";
+
+const invalid_stream_topic_regex = /[*>`]|(\$\$)/g;
+
+export function will_produce_broken_stream_topic_link(word: string): boolean {
+    return invalid_stream_topic_regex.test(word);
+}
+
+function get_stream_name_from_topic_link_syntax(syntax: string): string {
+    const start = syntax.indexOf("#**");
+    const end = syntax.lastIndexOf(">");
+    return syntax.slice(start + 3, end);
+}
+
+export function escape_invalid_stream_topic_characters(text: string): string {
+    switch (text) {
+        case "`":
+            return "&grave;";
+        case ">":
+            return "&gt;";
+        case "*":
+            return "&#42;";
+        case "$$":
+            return "&#36;&#36;";
+        default:
+            return text;
+    }
+}
```

Comment:
It cannot ever be correct to escape text using an escape character, without any arrangement for escaping the escape character itself!